x-pack/plugin/core/src/main/java/org/elasticsearch/protocol/xpack/XPackInfoResponse.java

```diff
@@ -203,11 +205,28 @@ public int hashCode() {
 
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-            builder.startObject()
-                .field("uid", uid)
-                .field("type", type)
-                .field("mode", mode)
-                .field("status", status.label());
+            builder.startObject();
+            builder.field("uid", uid);
+
+            final boolean acceptEnterprise = params.paramAsBoolean("accept_enterprise", false);
+            if (builder.getRestApiVersion() == RestApiVersion.V_7 && acceptEnterprise == false) {
```

Comment:
Nit: I'd inline this variable
```suggestion
            if (builder.getRestApiVersion() == RestApiVersion.V_7 && params.paramAsBoolean("accept_enterprise", false) == false) {
```
It reads better to me because the default value for `accept_enterprise` is different depend on `RestApiVersion`, i.e. it is `false` for `V7` while it is `true` for `V8`. So having it inlined after the `RestApiVersion` check is technically more accurate and it also avoid it being accidentally used in somewhere that is *not* guarded by the `RestApiVersion` check.